File: PHP/TimeNeededToBuyTickets.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

class TimeNeededToBuyTickets {    

    /**
     * @param Integer[] $tickets
     * @param Integer $k
     * @return Integer
     */
    function timeRequiredToBuy($tickets, $k) {
        $ans = 0;
        for ($i = 0; $i < count($tickets); $i++) {
            if ($i <= $k) {
                $ans += min($tickets[$k], $tickets[$i]);
            } else {
                $ans += min($tickets[$k] - 1, $tickets[$i]);
            }
        }
        return $ans;
    }

}

File: GO/TimeNeededToBuyTickets.go
/**
 * @version GO 1.22.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

func timeRequiredToBuy(tickets []int, k int) int {
	ans := 0
	for i := 0; i < len(tickets); i++ {
		if i <= k {
			ans += minTime(tickets[k], tickets[i])
		} else {
			ans += minTime(tickets[k]-1, tickets[i])
		}
	}
	return ans
}

func minTime(a, b int) int {
	if a < b {
		return a
	}
	return b
}
